hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestSecondaryIndex.scala

```diff
@@ -0,0 +1,222 @@
+/*
```

Comment:
Please ignore the repetive/duplicated code blocks. Will change once the functionality is ready and subsequently add more tests.